frontend/src/components/route/RoutePreview.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import styled from "styled-components/macro";
+
+export default function RoutePreview({ legs }) {
+
+    return (
+        <>
+            <ListStyled>
+                {legs.length === 0 ? null : <li><BoldSpan>Start</BoldSpan>{legs[0]?.startLatitude} / {legs[0]?.startLongitude}</li>}
+                {legs?.map(leg =>
+                    legs.indexOf(leg) + 1 === legs.length ?
+                    <li key={legs.indexOf(leg)}><BoldSpan>End</BoldSpan>{leg.endLatitude} / {leg.endLongitude}</li>
+                    :
+                    <li key={legs.indexOf(leg)}><BoldSpan>{legs.indexOf(leg) + 1}. Waypoint</BoldSpan>{leg.endLatitude} / {leg.endLongitude}</li>
```

Comment:
I'd suggest to replace `index.Of()` by simply calling the map function with the index:
`{legs?.map((leg, index) => index + 1 ...`

Reply:
Very good Idea! 😄